File: benchmarks/suite/alloc.go
package main

import "fmt"

type Node struct{ x, y int }

func main() {
	N, sink := 1_000_000, 0
	for i := 0; i < N; i++ {
		n := &Node{i, i + 1}
		sink += n.x
	}
	fmt.Println(sink)
}

File: benchmarks/suite/alloc.py
class Node:
    __slots__ = ('x', 'y')
    def __init__(self, a, b):
        self.x = a
        self.y = b

N = 1_000_000
sink = 0
for i in range(N):
    n = Node(i, i + 1)
    sink += n.x
print(sink)
